src/connectionController.ts

```diff
@@ -476,7 +476,7 @@ export default class ConnectionController {
     );
 
     if (!this._activeDataService) {
-      vscode.window.showErrorMessage(
+      void vscode.window.showErrorMessage(
```

Comment:
A lot of these `void` operators are in async functions … is there a reason to avoid using `await`? What happens if this fails?

Reply:
My 2 cents:

Personally I would default to awaiting every promise and only ignore (void) very specific ones for specific reasons. And then probably add a comment where it is ignored explaining the reasoning. Because if you await a promise it is clear where thrown errors go and bubble up and get caught. If not you end up with an unhandled rejection.

An example from my past of where code deliberately did _not_ await a promise was in code that kicked off some long-running "background" task inside a request handler and then responded with a 200 status code immediately. It still had mechanisms to catch and report errors, though - the returned promise was just handy in terms of testing so tests could wait for the thing to complete. But in that case the line of code had a comment above it explaining what was going on for future readers.

I suppose for these window object methods that return promises if the promises only resolved much later (like once the user closes a dialog or something), then that would be a good reason to ignore them. And if you have loads of calls to the same thing you wouldn't comment every time because it would be a common pattern in the project. I don't know if that's the case here, though.

Reply:
I had the same question. But I took a look at how vscode uses them and vscode extension samples and they do not await those 🤷🏻‍♀️

For example: https://github.com/microsoft/vscode-extension-samples/blob/main/terminal-sample/src/extension.ts#L231

the same with showInformationMessage: https://github.com/microsoft/vscode-extension-samples/blob/main/tree-view-sample/src/nodeDependencies.ts#L34

Reply:
The promise is resolved when modal is closed, so as we discussed in slack `then void is probably the right thing`.